File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        
        for (String word : babbling){
            
            if (isValidWord(word)){
                answer++;
            }
        }
        return answer;
    }
    
    private boolean isValidWord(String word){
    String[] validSounds = { "aya", "ye", "woo", "ma"};
    
    for(String sound : validSounds){
        word = word.replaceFirst(sound, "-");
    }
    
    return word.replaceAll("-", "").isEmpty();
}
}

File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.py
from itertools import permutations

def solution(babbling):
    speak = ["aya", "ye", "woo", "ma"]
    possible = set()
    
    for i in range(1, 5):
        for p in permutations(speak, i):
            possible.add("".join(p))
    
            
    return sum(1 for b in babbling if b in possible)